### PR TITLE
Update url.py

### PR DESCRIPTION
**Pull request by srisankethu**
Just made a few changes like in line 104,105 is nested under  if statement in line 102

if port is not None: statement in line 100 is added to else: statement in line 111

Combined url=... from line 99 and 120

### CODE REVIEW

**Comment by sontek** [comment]
Can you add a test to show why this change was necessary?

**Comment by srisankethu** [comment]
I was just organising the code.
Nested if on line 104 looks meaningful
No need to update url twice(line 99 and line 120) as there is no change in it's value between line 99 and line 120
Unnecessary if port is None: on line 100 as else statement on line 110 is the same.

File: pyramid/url.py [hunk 1]
```diff
@@ -96,19 +96,17 @@ def _partial_application_url(self, scheme=None, host=None, port=None):
...
-            host = e['SERVER_NAME']
+            if host is None:
+                host = e['SERVER_NAME']
         if port is None:
             if ':' in host:
                 host, port = host.split(':', 1)
             else:
                 port = e['SERVER_PORT']
         else:
+            port = str(port)
```
Comment (stevepiercy) [inline pyramid/url.py]: Why is this line necessary? Please provide a test to prove its necessity.

File: pyramid/url.py [hunk 2]
```diff
@@ -96,19 +96,17 @@ def _partial_application_url(self, scheme=None, host=None, port=None):
...
                     port = '80'
-        url = scheme + '://'
-        if port is not None:
-            port = str(port)
         if host is None:
             host = e.get('HTTP_HOST')
-        if host is None:
-            host = e['SERVER_NAME']
+            if host is None:
+                host = e['SERVER_NAME']
```
Comment (stevepiercy) [inline pyramid/url.py]: This is OK with me. With the nesting, I read this implicitly as "if host is *still* None, then...".

File: pyramid/url.py [hunk 3]
```diff
@@ -117,7 +115,7 @@ def _partial_application_url(self, scheme=None, host=None, port=None):
         elif scheme == 'http':
             if port == '80':
                 port = None
-        url += host
+        url = scheme + '://' + host
```
Comment (stevepiercy) [inline pyramid/url.py]: Condensing this url assignment into one line from the previous two lines (old lines 99 and 120) is cleaner. :+1:

File: pyramid/url.py (hunk 1 above)
Comment (srisankethu) [inline pyramid/url.py]: That was already a part of your code.I just rearranged it's place

Comment (stevepiercy) [inline pyramid/url.py]: Ah, I see now. I'm OK with this change, too.

**Comment by stevepiercy** [comment]
If one of the core contributors gives a :+1: on this PR, then I'll merge it, backport it to 1.6 and 1.5, and make a change note.

@srisankethu would you please sign CONTRIBUTORS.txt, and push it to this PR, too? You don't need to delete what you've already done.

**Comment by mmerickel** [comment]
There's no need to backport this change, it's just a cleanup versus adding any actual features.

**Comment by mmerickel** [comment]
Anyway, +1, I'll merge this and add a changelog entry. Thanks!

**Comment by mmerickel** [comment]
I lied... need you to add your name to CONTRIBUTORS.txt first. :-)

**Comment by srisankethu** [comment]
Ha....Thanks man
